src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -843,7 +865,7 @@ private BetterResult BetterOperator(BinaryOperatorSignature op1, BinaryOperatorS
             Debug.Assert(op1.Priority.HasValue == op2.Priority.HasValue);
 
             // We use Priority as a tie-breaker to help match native compiler bugs.
-            if (op1.Priority.HasValue && op2.Priority.HasValue && op1.Priority.GetValueOrDefault() != op2.Priority.GetValueOrDefault())
+            if (op1.Priority.HasValue && op1.Priority.GetValueOrDefault() != op2.Priority.GetValueOrDefault())
```

Comment:
`op2.Priority.HasValue` was just redundant since we have asserted that `Priority.HasValue` is equal between both operands. I can exclude this change if you prefer.